pulp_smash/tests/rpm/api_v2/test_no_op_publish.py

```diff
@@ -0,0 +1,220 @@
+* `Pulp Smash #232 <https://github.com/PulpQE/pulp-smash/issues/232>`_
+"""
+from __future__ import unicode_literals
+
+import inspect
+
+import unittest2
+from packaging.version import Version
+
+from pulp_smash import api, config, exceptions, utils
+from pulp_smash.compat import urljoin
+from pulp_smash.constants import REPOSITORY_PATH, RPM_FEED_URL
+from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.rpm.utils import set_up_module
+
+
+_REPO = None
+"""A dict of information about the repository created by ``setUpModule``."""
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Possibly skip the tests in this module. Create and sync an RPM repo.
+
+    Skip this module of tests if Pulp is older than version 2.9. (See `Pulp
+    #1724`_.) Then create an RPM repository with a feed and sync it. Test cases
+    may copy data from this repository but should **not** change it.
+
+    .. _Pulp #1724: https://pulp.plan.io/issues/1724
+    """
+    set_up_module()
+    cfg = config.get_config()
+    if cfg.version < Version('2.9'):
+        raise unittest2.SkipTest('This module requires Pulp 2.9 or greater.')
+
+    # Create and sync a repository. If this set-up procedure grows, consider
+    # implementing a stack of tear-down actions
+    client = api.Client(cfg, api.json_handler)
+    body = gen_repo()
+    body['importer_config']['feed'] = RPM_FEED_URL
+    global _REPO  # pylint:disable=global-statement
+    _REPO = client.post(REPOSITORY_PATH, body)
+    try:
+        utils.sync_repo(cfg, _REPO['_href'])
+    except (exceptions.CallReportError, exceptions.TaskReportError,
+            exceptions.TaskTimedOutError):
+        client.delete(_REPO['_href'])
+        raise
+
+
+def tearDownModule():  # pylint:disable=invalid-name
+    """Delete the repository created by :meth:`setUpModule`."""
+    api.Client(config.get_config()).delete(_REPO['_href'])
+
+
+def get_repomd_xml_path(distributor_rel_url):
+    """Construct the path to a repository's ``repomd.xml`` file.
+
+    :param distributor_rel_url: A distributor's ``relative_url`` option.
+    :returns: An string path to a ``repomd.xml`` file.
+    """
+    path = urljoin('/pulp/repos/', distributor_rel_url)
+    return urljoin(path, 'repodata/repomd.xml')
+
+
+class BaseTestCase(utils.BaseAPITestCase):
+    """Provide common behaviour for the test cases in this module."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a repository with a distributor, and populate it.
+
+        In addition, create several variables for use by the test methods.
+        """
+        if inspect.getmro(cls)[0] == BaseTestCase:
+            raise unittest2.SkipTest('Abstract base class.')
+        super(BaseTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        repo_href = client.post(REPOSITORY_PATH, gen_repo())['_href']
+        cls.resources.add(repo_href)  # mark for deletion
```

Comment:
Would be great if these test classes have a method like `cls.mark_for_deletion(repo_href)`, just brainstorming.